docs/reference/hooks.rst

```diff
@@ -2,3 +2,107 @@
 
 Hooks
 =====
+
+A hook is an executable file that runs within a snap’s confined environment when
+a certain action occurs.
+
+Common scenarios requiring hooks include:
+
+- Notifying a snap that something has happened. For example, if a snap has been
+  upgraded, the snap may need to trigger a scripted migration process to port an
+  old data format to the new one.
+
+- Notifying a snap that a specific operation is in progress. For example, a snap may
+  need to know when a specific interface connects or disconnects.
+
+Transactions and rollback
+-------------------------
+
+Snapd executes a hook as a single transaction, where a transaction holds all the
+configuration changes made by the hook. These changes are atomic and are only
+visible to the running system after the hook finishes executing.
+
+This allows changes to be rolled back if errors occur during the execution of a hook.
+This happens if a non-zero value is returned with either the configure or
+default-configure hooks or if an error occurs with any hook involved with
+an `interface auto-connection <https://snapcraft.io/docs/auto-connection-mechanism>`_.
+
+Environment
+-----------
+
+If the hook executable is a shell script, it should assume a POSIX-compliant shell
+environment for its execution.
+
+If your script needs a specific interpreter, such as Bash or Python, it needs to be
+explicitly declared within the script’s hashbang header (``#!/bin/bash``, for example)
+and your snap needs to ensure the interpreter is available.
+
+Hook names
+----------
+
+The filename of the executable should be the name of the hook. If the file
+exists, snapd will execute the file when required by that hook’s action. See
+`Supported snap hooks <https://snapcraft.io/docs/supported-snap-hooks>`_
```

Comment:
I think we're retiring the rest of this content, so we shouldn't be linking to this page.

Is there another way to find all the available hooks? If not, we should reconsider and include the compendium of hooks on this page.

Reply:
There's nowhere else where each hook is described, so we should do something with them.

I think it's most appropriate to put them in snapd's side of the docs.  The list of supported hooks and the behavior of each hook at runtime is entirely driven by snapd.  

Snapcraft is concerned with the hook's name, location, and permissions of the hooks at build-time (which is what we have documented in our RTD pages).

Reply:
Got it. I'll add it to the snapd migration tasks.